azurecontainerapps.ts

```diff
@@ -170,6 +178,20 @@ export class azurecontainerapps {
             this.toolHelper.writeError(conflictingArgumentsMessage);
             throw Error(conflictingArgumentsMessage);
         }
+
+        // Ensure that the imageToBuild and source arguments are both provided and imageToBuild starts with 'default/'
+        if (!this.util.isNullOrEmpty(this.imageToBuild) && !this.util.isNullOrEmpty(this.appSourcePath) && !this.imageToBuild.startsWith('default/')) {
```

Comment:
I'm not sure I understand this check -- if the user wants to use any non-internal registry flow, this validation automatically breaks them. For example, if they provide their application source with a Dockerfile, and they provide a value for `imageToBuild` that tells us "I want to build the Dockerfile in my provided application source with this image name", we're going to fail them here. Same goes for basically every non-internal registry flow where the user tells us what they want their image name to be.

Reply:
Ok, this condition was meant to be `if (!this.util.isNullOrEmpty(this.imageToBuild) && !this.util.isNullOrEmpty(this.appSourcePath) && !this.imageToBuild.startsWith('default/')`. 
Didn't realize there was a ! after `this.imageToBuild.startsWith('default/')`.

This condition makes sure that the `appSourcePath` arg is provided for the internal registry scenario.

Reply:
You'll also want to remove the `!` before the check for `appSourcePath` because right now the conditional is equivalent to "if the user provides an application source path AND they provide an image to build that begins with `default/`, throw an exception"

Reply:
I missed that. That you!